### PR TITLE
rpc: add tests for simulate transaction inner instructions

### PR DESCRIPTION
**Pull request by buffalojoec**
#### Problem
As of #34313, inner instructions can now optionally be included in the RPC response for `simulateTransaction`. However, the test coverage on this new feature is lacking.

#### Summary of Changes
Added some tests specifically for the `innerInstructions` on the RPC's `simulateTransaction` method.

### CODE REVIEW

**Comment by buffalojoec** [comment]
Ah nuts, I think some recent Bank changes nuked this test setup.

**Comment by buffalojoec** [comment]
This should be ready for another look @CriesofCarrots.

Is there a way to trigger CI manually for PRs? I think the "clippy on MacOS" job hiccuped, but I seem to be compliant here.

I can push a rebase sometime in the future as well to trigger if need be.

**Comment by steveluscher** [comment]
So excited for this! If you get a chance to update [the docs](https://github.com/solana-foundation/developer-content/blob/e3ab72171725e4089be130f4a481288533e59f0a/docs/rpc/http/simulateTransaction.mdx#L100) too… 🤌🏻

**Comment by buffalojoec** [comment]
I rewrote this to use built-in program instructions to avoid the added dependencies and setup stuff. Much cleaner now.

If we want coverage on the token program instructions, etc. We'll have to consider adding `program-test` or at least the `.so` files of the SPL programs we want to test.

File: rpc/src/rpc.rs [hunk 1]
```diff
@@ -6294,6 +6294,298 @@ pub mod tests {
...
+            },
+            "id": 1,
+        });
+        let expected: Response =
+            serde_json::from_value(expected).expect("expected response deserialization");
+        let result: Response = serde_json::from_str(&res.expect("actual response"))
+            .expect("actual response deserialization");
+        assert_eq!(result, expected);
+
+        // Try some other instructions
```
Comment (CriesofCarrots) [inline rpc/src/rpc.rs:6503]: What's the value of this additional test case?

**Review by CriesofCarrots — COMMENTED** [review]
Nice, much lighter, thanks!
Just a couple suggestions for simplification, and one question

File: rpc/src/rpc.rs (hunk 1 above)
Comment (buffalojoec) [reply rpc/src/rpc.rs:6503]: I just figured trying it on more than one instruction would be good. I can take it out, though if it's not necessary.

Comment (CriesofCarrots) [reply rpc/src/rpc.rs:6505]: Yeah, I think we can take it out.

**Review by CriesofCarrots — COMMENTED** [review]
r+ once CI is happy; you'll need to rebase to pick up the audit fix

**Comment by CriesofCarrots** [comment]
And yes, please update the JSON-RPC docs! 🙏 (I know I'm going to have a hard time remembering to update them now that they're out of this repo. Hopefully we can help each other :))

**Comment by buffalojoec** [comment]
> And yes, please update the JSON-RPC docs! 🙏 (I know I'm going to have a hard time remembering to update them now that they're out of this repo. Hopefully we can help each other :))

I put it on my list!

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**